Source/player.cpp

```diff
@@ -2447,68 +2447,27 @@ void NewPlrAnim(Player &player, player_graphic graphic, Direction dir, int numbe
 void SetPlrAnims(Player &player)
 {
 	HeroClass pc = player._pClass;
+	auto gn = static_cast<PlayerWeaponGraphic>(player._pgfxnum & 0xF);
 
 	if (leveltype == DTYPE_TOWN) {
-		player._pNFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][6];
-		player._pWFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][7];
-		player._pDFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][3];
-		player._pSFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][4];
+		player._pNFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][7];
+		player._pWFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][8];
+		player._pDFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][4];
+		player._pSFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][5];
 	} else {
 		player._pNFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][0];
-		player._pWFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][1];
-		player._pHFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][5];
-		player._pSFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][4];
-		player._pDFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][3];
-		player._pBFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][2];
+		player._pWFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][2];
+		player._pAFrames = PlrGFXAFrames[static_cast<std::size_t>(pc)][static_cast<std::size_t>(gn)];
+		player._pHFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][6];
+		player._pSFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][5];
+		player._pDFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][4];
+		player._pBFrames = PlrGFXAnimLens[static_cast<std::size_t>(pc)][3];
+		player._pAFNum = PlrGFXAFNum[static_cast<std::size_t>(pc)][static_cast<std::size_t>(gn)];
 	}
-	player._pSFNum = PlrGFXAnimLens[static_cast<std::size_t>(pc)][8];
+	player._pSFNum = PlrGFXAnimLens[static_cast<std::size_t>(pc)][10];
 
-	auto gn = static_cast<PlayerWeaponGraphic>(player._pgfxnum & 0xF);
 	int armorGraphicIndex = player._pgfxnum & ~0xF;

```

Comment:
I would move this to line 2451, to be consistent.

Reply:
then you have a mess where you don't see instantly where each variable is being used.